test-benchmark.html

```diff
@@ -0,0 +1,431 @@
+```
+
+Every `template` tag within the `test-benchmark` element will be transformed
+into a test that will be considered within the broader benchmark.
+
+### Imperative configuration
+
+Tests can be configured imperatively by using the `addTest` method. A test's
+name, HTML content and optional list of tags should be provided. The HTML
+content provided should reflect the content that would have otherwise been
+found in a corresponding test temmplate:
+
+```js
+var benchmark = Polymer.dom(document).querySelector('test-benchmark');
+var html = '<script>\
+StartTest();\
+// Do some work...\
+EndTest();\
+<\/script>';
+
+benchmark.addTest('Test one', html);
+```
+
+## Anatomy of a test run
+
+In order to run a test and collect data, `test-benchmark` performs the following
+steps:
+
+ 1. Wrap the provided test HTML in a light weight test harness. This harness
+    brings in the Web Components polyfill and also craetes the global
+    `StartTest` and `EndTest` functions. test HTML is inserted into the
+    `<body>` tag of the test harness.
+ 2. Convert the wrapped test HTML into a [Blob][1], and then [create an object
+    URL][0] from that blob.
+ 3. Assign the object URL as the `src` attribute of an `<iframe>`.
+ 4. When `EndTest` is called, a [message is posted][2] from the child `<iframe>`
+    to the parent frame indicating that the test ended and providing the time
+    between `StartTest` and `EndTest` as measured by
+    [`window.performance.now()`][3].
+
+## Browser limitations
+
+### Inert templates
+
+It may be desirable to create test benchmarks in a document root using the
+declarative configuration style in browsers that do not support inert template
+tags. The Web Components polyfill will give you inert template tags in HTML
+imports, but *not* in the main document. To offer a partial work-around, one
+may mark script tags with `type="benchmark"` in order to prevent scripts from
+executing. This attribute will be removed from the script before the test is
+created.
+
+### Required features
+
+This element depends on the ability to use a [data URI][4] as the `src`
+attribute for an `<iframe>` element. This feature is supported by pretty much
+every browser. *However*, no known version of Microsoft's browsers - including
+Edge - support this feature as of this writing.
+
+`test-benchmark` performs a feature test for the ability to use [data URIs][4]
+as described above, and will not perform test runs in browsers that do not
+support the necessary feature.
+
+[0]: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL
+[1]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+[2]: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
+[3]: https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+[4]: http://caniuse.com/#feat=datauri
+
+@demo demo/index.html
+-->
+
+<dom-module id="test-benchmark">
+  <template>
+    <style>
+      iframe {
+        position: absolute;
+        clip: rect(0, 0, 0, 0);
```

Comment:
I was worried that `display: none` may have an adverse affect on things like benchmarking render.